hydradx/tests/test_arbitrage_agent.py

```diff
@@ -529,29 +700,169 @@ def test_max_liquidity(dotusd_price_mult: float, dot_amts: list, max_trade: floa
 
+    hdxusd_spot = op_state.price(op_state, 'HDX', 'USDT')
+    hdxusd_spot_adj = hdxusd_spot * hdxusd_price_mult
+
+    hdx_usdt_order_book = {
+        'bids': [{'price': hdxusd_spot_adj * 0.999, 'amount': hdxusd_amts[0]},
+                 {'price': hdxusd_spot_adj * 0.99, 'amount': hdxusd_amts[1]},
+                 {'price': hdxusd_spot_adj * 0.9, 'amount': hdxusd_amts[2]},
+                 {'price': hdxusd_spot_adj * 0.8, 'amount': hdxusd_amts[3]}],
+        'asks': [{'price': hdxusd_spot_adj * 1.001, 'amount': hdxusd_amts[4]},
+                 {'price': hdxusd_spot_adj * 1.01, 'amount': hdxusd_amts[5]},
+                 {'price': hdxusd_spot_adj * 1.1, 'amount': hdxusd_amts[6]},
+                 {'price': hdxusd_spot_adj * 1.2, 'amount': hdxusd_amts[7]}]
+    }
+
+    hdx_usdt_order_book_obj = OrderBook([[bid['price'], bid['amount']] for bid in hdx_usdt_order_book['bids']],
+                                        [[ask['price'], ask['amount']] for ask in hdx_usdt_order_book['asks']])
+
+    hdxdot_spot = op_state.price(op_state, 'HDX', 'DOT')
+    hdxdot_spot_adj = hdxdot_spot * hdxdot_price_mult
+
+    hdx_dot_order_book = {
+        'bids': [{'price': hdxdot_spot_adj * 0.99, 'amount': hdxdot_amts[0]},
+                 {'price': hdxdot_spot_adj * 0.9, 'amount': hdxdot_amts[1]}],
+        'asks': [{'price': hdxdot_spot_adj * 1.01, 'amount': hdxdot_amts[2]},
+                 {'price': hdxdot_spot_adj * 1.1, 'amount': hdxdot_amts[3]}]
+    }
+
+    hdx_dot_order_book_obj = OrderBook([[bid['price'], bid['amount']] for bid in hdx_dot_order_book['bids']],
+                                       [[ask['price'], ask['amount']] for ask in hdx_dot_order_book['asks']])
+
     order_book = {
-        ('DOT', 'USDT'): dot_usdt_order_book_obj
+        ('DOT', 'USDT'): dot_usdt_order_book_obj,
+        ('HDX', 'USDT'): hdx_usdt_order_book_obj,
+        ('HDX', 'DOT'): hdx_dot_order_book_obj
     }
 
+    cfg = [{"tkn_pair": k, "exchange": "exchange_name", "order_book": k, "buffer": buffer_ls[i]} for i, k in enumerate(order_book)]
+
     cex = CentralizedMarket(
         order_book=order_book,
         asset_list=['USDT', 'DOT', 'HDX'],
         trade_fee=cex_fee
     )
 
-    order_book_map = {k: k for k in order_book}
+    arb_swaps = get_arb_swaps_simple(op_state, {'exchange_name': cex}, cfg)
+    initial_agent = Agent(holdings={'USDT': 1000000000, 'DOT': 1000000000, 'HDX': 1000000000}, unique_id='bot')
+    agent = initial_agent.copy()
 
-    arb_swaps = get_arb_swaps(op_state, cex, order_book_map, max_trades={('DOT', 'USDT'): max_trade})
+    execute_arb(op_state, {'exchange_name': cex}, agent, arb_swaps)
 
-    swap_list = arb_swaps[('DOT', 'USDT')]
-    total_amt = sum([swap['cex']['amount'] for swap in swap_list])
-    if total_amt > max_trade:
-        raise
+    profit = calculate_profit(initial_agent, agent)
+    for tkn in profit:
+        if profit[tkn] / initial_agent.holdings[tkn] < -1e-10:
```

Comment:
that's true but unfortunately because of the way that Python handles precision, how arbitrarily large these numbers are determines the precision of the results